server/src/main/java/org/opensearch/rest/action/admin/cluster/RestDeleteDecommissionStateAction.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rest.action.admin.cluster;
+
+import org.opensearch.action.admin.cluster.decommission.awareness.delete.DeleteDecommissionStateRequest;
+import org.opensearch.client.Requests;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.action.RestToXContentListener;
+
+import java.io.IOException;
+import java.util.List;
+
+import static java.util.Collections.singletonList;
+import static org.opensearch.rest.RestRequest.Method.DELETE;
+
+/**
+ * Clears the decommission metadata.
+ *
+ * @opensearch.api
+ */
+public class RestDeleteDecommissionStateAction extends BaseRestHandler {
+
+    @Override
+    public List<Route> routes() {
+        return singletonList(new Route(DELETE, "/_cluster/decommission/awareness"));
+    }
+
+    @Override
+    public String getName() {
+        return "delete_decommission_state_action";
+    }
+
+    @Override
+    protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
+        DeleteDecommissionStateRequest deleteDecommissionStateRequest = createRequest();
+        return channel -> client.admin()
+            .cluster()
+            .deleteDecommissionState(deleteDecommissionStateRequest, new RestToXContentListener<>(channel));
+    }
```

Comment:
If there are multiple attributes we should just clear only that attribute. Hope that has been factored in

Reply:
Currently we are deleting the decommissionAttribute of any type which is set. Also as per now only one decommission attribute can be set. So we are clearing that one only.
Creating a github issue to handle specific attribute. Will add the Github issue here.

Reply:
Create an issue
https://github.com/opensearch-project/OpenSearch/issues/4654